org-timeline.el

```diff
@@ -147,25 +188,55 @@ Return new copy of STRING."
                           current-time))
                    (face (org-timeline--get-face)))
               (when (>= beg start-offset)
-                (push (list beg end face) tasks)))))))
-    (setq tasks (nreverse tasks))
+                (push (list beg end face txt line) tasks)))))))
+    (nreverse tasks)))
+
+(defun org-timeline--generate-timeline ()
+  "Generate the timeline string that will represent current agenda view."
+  (let* ((start-offset (* org-timeline-start-hour 60))
+         (current-time (+ (* 60 (string-to-number (format-time-string "%H")))
+                          (string-to-number (format-time-string "%M"))))
+         (current-offset (/ (- current-time start-offset) 10))
+         (slotline (org-timeline--add-elapsed-face
+                    "|     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |"
+                    current-offset))
+         (hourline (org-timeline--add-elapsed-face
+                    (concat "|"
+                            (mapconcat (lambda (x) (format "%02d:00" (mod x 24)))
+                                       (number-sequence org-timeline-start-hour (+ org-timeline-start-hour 23))
+                                       "|")
+                            "|")
+                    current-offset))
+         (timeline (concat hourline "\n" slotline))
+         (tasks (org-timeline--list-tasks)))
     (cl-labels ((get-start-pos (current-line beg) (+ 1 (* current-line (1+ (length slotline))) (/ (- beg start-offset) 10)))
                 (get-end-pos (current-line end) (+ 1 (* current-line (1+ (length slotline))) (/ (- end start-offset) 10))))
       (let ((current-line 1))
         (with-temp-buffer
           (insert timeline)
           (-each tasks
-            (-lambda ((beg end face))
+            (-lambda ((beg end face txt line))
               (while (get-text-property (get-start-pos current-line beg) 'org-timeline-occupied)
                 (cl-incf current-line)
                 (when (> (get-start-pos current-line beg) (point-max))
                   (save-excursion
                     (goto-char (point-max))
                     (insert "\n" slotline))))
               (let ((start-pos (get-start-pos current-line beg))
-                    (end-pos (get-end-pos current-line end)))
-                (put-text-property start-pos end-pos 'font-lock-face face)
-                (put-text-property start-pos end-pos 'org-timeline-occupied t))
+                    (end-pos (get-end-pos current-line end))
+                    (move-to-task-map (make-sparse-keymap))
```

Comment:
This could probably be defined only once outside of the loop, along with the bindings.